### PR TITLE
 DOC: Fix critical value significance levels in stats.anderson_ksamp

### LINKED ISSUE #10263: Documentation error for K-sample Anderson test
The[ documentation for the k sample Anderson test](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.anderson_ksamp.html) has incorrect values for the significance levels associated with the returned critical values.

Currently the documentaion shows:
`    critical_values : array
        The critical values for significance levels 25%, 10%, 5%, 2.5%, 1%.`

Where as the the code in morestats.py on line 2039 shows:
`sig = np.array([0.25, 0.1, 0.05, 0.025, 0.01, 0.005, 0.001])`

There is also an issue with the signifance_level as the value is not floored at 1%, but at 0.1% as outlined in the notes.

I am using scipy version 1.2. However these errors are present in the version 1.3 docs as well.

### PR DESCRIPTION
**Pull request by jmmanley**
Fixes error in documentation for k-sample Anderson test pointed out by @amulliner-dstl in #10263.

My first PR, hoping to contribute lots more! :)

Fixes #10263

### CODE REVIEW

File: scipy/stats/morestats.py [hunk 1]
```diff
@@ -1919,7 +1919,8 @@ def anderson_ksamp(samples, midrank=True):
     statistic : float
         Normalized k-sample Anderson-Darling test statistic.
     critical_values : array
-        The critical values for significance levels 25%, 10%, 5%, 2.5%, 1%.
+        The critical values for significance levels 25%, 10%, 5%, 2.5%, 1%,
+        0.5%, 0.1%.
```
Comment (tylerjereddy) [inline scipy/stats/morestats.py]: This looks good, but I think the documented floor of the `significance_level` mentioned below also needs to be updated now.

That is also mentioned in the linked issue. Once that is fixed, I think we can properly say that this "Fixes" the linked issue & have it automatically close once this is merged.

Comment (jmmanley) [reply scipy/stats/morestats.py]: Thanks for the catch!

**Review by chrisb83 — APPROVED** [review]
Looks good, thanks @tylerjereddy for the review

**Comment by chrisb83** [comment]
@jmmanley thanks for fixing this, I forgot to update the docstring when I extended the values

**Pull request merged**